### PR TITLE
SB-868 - Announcement attachment up loader refactoring with injectable service/factory

### PR DESCRIPTION
**Pull request by nilesh-more**
- [x] [ Announcement attachment up loader refactoring with injectable service/factory](https://project-sunbird.atlassian.net/browse/SB-868?filter=-1)
- [x] style doc added
- [x] gulp test - done

### CODE REVIEW

**Comment by harishgilimi** [comment]
closing PR for cleaning repo

File: src/app/private/scripts/controllers/announcement/createAnnouncementController.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict'
-angular.module('playerApp').controller('createAnnouncementCtrl', ['$rootScope', '$scope', '$timeout', 'config', 'toasterService', 'announcementService', '$filter', 'uuid4',
-  function ($rootScope, $scope, $timeout, config, toasterService, announcementService, $filter, uuid4) {
+angular.module('playerApp').controller('createAnnouncementCtrl', ['$rootScope', '$scope', '$timeout', 'config', 'toasterService', 'announcementService', '$filter', 'uuid4', 'fileUpload',
+  function ($rootScope, $scope, $timeout, config, toasterService, announcementService, $filter, uuid4, fileUpload) {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/announcement/createAnnouncementController.js]: $services should be placed before others, please move $filter.

File: src/app/private/scripts/controllers/announcement/createAnnouncementController.js [hunk 2]
```diff
@@ -131,6 +170,13 @@ angular.module('playerApp').controller('createAnnouncementCtrl', ['$rootScope',
...
       }
     }
+
+    /**
+     * @method removeRicipients
+     * @desc - remove selected recipients
+     * @memberOf Controllers.createAnnouncementCtrl
+     * @param {object} [item] [current selected item]
+     */
     createAnn.removeRicipients = function (item) {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/announcement/createAnnouncementController.js]: sp: ricipients > recipients

File: src/app/private/scripts/factories/fileUploadFactory.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  .factory('fileUpload', ['$timeout', 'toasterService', function ($timeout, toasterService) {
+    return {
+      /**
+       * @method initializeFineUploader
+       * @desc initialize FineUploader third party plugin
+       * @memberOf Factory.fineUploader
+       * @param {Object}  option - Option object to invoke controller callback function
+       * @returns {Callback} Trigger callback function
+       */
+      createFineUploadInstance: function (options) {
```
Comment (kochhar) [inline src/app/private/scripts/factories/fileUploadFactory.js]: having the uploader type name in the method does not prevent strong coupling. having the options sent from the controller doesn't prevent abstraction leak.

the factory method should be called when the app is initialised. app-level config options should be passed in and an instance created. this instance is available as a partially configured factory. this factory can then be called with controller specific options to get the uploader instance.

File: src/test/spec/app/private/scripts/controllers/announcement/createAnnouncementController.js [hunk 4]
```diff
@@ -94,14 +94,14 @@ describe('Controller: createAnnouncementCtrl', function () {
   })
 
   it('convert file size into KB / MB', function (done) {
```
Comment (kochhar) [inline src/test/spec/app/private/scripts/controllers/announcement/createAnnouncementController.js]: missing expectations here...

File: src/test/spec/app/private/scripts/controllers/announcement/createAnnouncementController.js [hunk 5]
```diff
@@ -94,14 +94,14 @@ describe('Controller: createAnnouncementCtrl', function () {
...
+    spyOn(createAnn, 'getReadableFileSize').and.callThrough()
+    createAnn.getReadableFileSize(1234)
     done()
   })
 
   it('should not convert file size into KB / MB', function (done) {
-    spyOn(createAnn, 'convertFileSize').and.callThrough()
-    createAnn.convertFileSize(0)
+    spyOn(createAnn, 'getReadableFileSize').and.callThrough()
+    createAnn.getReadableFileSize(0)
```
Comment (kochhar) [inline src/test/spec/app/private/scripts/controllers/announcement/createAnnouncementController.js]: ... and here.

File: src/app/private/scripts/controllers/announcement/createAnnouncementController.js (hunk 1 above)
Comment (nilesh-more) [reply src/app/private/scripts/controllers/announcement/createAnnouncementController.js]: @kochhar resolved
All '$' item should be injected first. cc: @harishgilimi

File: src/app/private/scripts/controllers/announcement/createAnnouncementController.js (hunk 2 above)
Comment (nilesh-more) [reply src/app/private/scripts/controllers/announcement/createAnnouncementController.js]: @kochhar resolved

**Pull request merged**